pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -180,6 +182,30 @@ func updateIdentity(set *appsv1alpha1.StatefulSet, pod *v1.Pod) {
 	pod.Labels[apps.StatefulSetPodNameLabel] = pod.Name
 }
 
+// isRunningAndAvailable returns true if pod is in the PodRunning Phase,
+// and it has a condition of PodReady for a minimum of minReadySeconds.
+// return true if it's available
+// return false with zero means it's not ready
+// return false with a positive value means it's not available and should recheck with that time
+func isRunningAndAvailable(pod *v1.Pod, minReadySeconds int32) (bool, time.Duration) {
+	if pod.Status.Phase != v1.PodRunning || !podutil.IsPodReady(pod) {
+		return false, 0
+	}
+	c := podutil.GetPodReadyCondition(pod.Status)
+	minReadySecondsDuration := time.Duration(minReadySeconds) * time.Second
+	if minReadySeconds == 0 {
+		return true, 0
+	}
+	if c.LastTransitionTime.IsZero() {
```

Comment:
Is this possible to happen if the Pod status is ready but the transition time is 0?

Reply:
not sure, this case presents in the `IsPodReady` function in the  "k8s.io/kubernetes/pkg/api/v1/pod"